services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationDataMapper.kt

```diff
@@ -74,4 +81,16 @@ object ApplicationDataMapper {
             }
         }
     }
+
+    private fun path(vararg field: String) = field.joinToString(separator = ".", prefix = BASE)
+
+    private fun ZonedDateTime?.orThrow(path: String) = this ?: throw AlluDataException(path, NULL)
```

Comment:
👍 Nice addition.

Though if this was generic, it could also be used for `rockExcavation` in `workDescription()`.
```suggestion
    private fun <T> T?.orThrow(path: String) = this ?: throw AlluDataException(path, NULL)
```

Reply:
Fixed by using generics.